presto-main/src/main/java/com/facebook/presto/cost/FilterStatsCalculator.java

```diff
@@ -259,14 +294,21 @@ protected PlanNodeStatsEstimate visitComparisonExpression(ComparisonExpression n
 
             Optional<Symbol> leftSymbol = asSymbol(left);
             SymbolStatsEstimate leftStats = getExpressionStats(left);
+            if (Objects.equals(leftStats, UNKNOWN_STATS)) {
```

Comment:
Why do we return (implicitly) `Optional.empty()` here. Propagating unknown stats isn't the same as using some arbitrary value for filtering.

Reply:
So I see there is a distinction between unknown stats that originate from connector or unknown stats that because we cannot compute those. This case is the former one. Therefore we probably shouldn't bailout to `Optional.empty` here.

Reply:
if `leftState.distinctValues()` is `NaN` then this will be `false` and we will return `UNKNOWN_STATS`. Then in https://github.com/prestodb/presto/pull/10384/commits/92d70e10865dee1a7387ccce3119da2515dc565a#diff-2c8e7079274d1e50a57e44770854b2c5R140 will propagate unknown stats instead of using filtering constant.

Probably we shouldn't have double representation of "unknown_stats": `Optional.empty()` and `NaN`s. The latter was just fine.

Let's land it and think about it later.